### PR TITLE
Add API deprecation middleware

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

- Adds a new deprecation middleware to inform about deprecated API paths.
- Adds throttle logic to not spam the user with deprecation messages.

As a start, only `/hardware/trigger` is added to this, but all API paths that we have deprecated should use this as well.

Example logs:

```
WARNING (MainThread) [supervisor.api.deprecation] Access to deprecated API '/hardware/trigger' detected from deCONZ - please report this to the maintainer of the deCONZ add-on
```

```
WARNING (MainThread) [supervisor.api.deprecation] Access to deprecated API '/hardware/trigger' detected from deCONZ - you are currently running version 6.6.3, there is an update pending for 6.6.4. If that does not help please report this to the maintainer of the deCONZ add-on
```

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/deprecation.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                self.write_log(request)
+
+        return await handler(request)
+
+    def write_log(self, request: Request):
+        """Write the log entry."""
+        request_from = "unknown"
+        more_info = ""
+
+        supervisor_token = excract_supervisor_token(request)
```
Comment (pvizeli) [inline supervisor/api/deprecation.py:51]: You don't need do that, you can just access to the data before which other middleware have generated

Comment (pvizeli) [reply supervisor/api/deprecation.py:51]: Maybe just a simple function decorator would also do it instead to have it as middleware on each call?

Comment (ludeeus) [reply supervisor/api/deprecation.py:51]: Sure

File: supervisor/api/deprecation.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ) -> Response:
+        """Check if the requested path is deprecated."""
+        if request.path in DEPRECATED_PATHS:
+            now = datetime.datetime.now()
+            if request.path not in self._trottle or (
+                now - self._trottle[request.path]
+            ) > DEPRECATED_PATHS[request.path].get(
+                "grace_period", DEFAULT_GRACE_PERIOD
+            ):
+                self._trottle[request.path] = now
```
Comment (ludeeus) [inline supervisor/api/deprecation.py:45]: This should be pr. path and request_from, if you have multiple addons, it should give multiple messages

**Pull request closed without merging**